sqlglot/dialects/tsql.py

```diff
@@ -370,6 +370,17 @@ def _timestrtotime_sql(self: TSQL.Generator, expression: exp.TimeStrToTime):
     return sql
 
 
+def _parse_datetrunc(args: t.List) -> exp.TimestampTrunc:
+    unit = seq_get(args, 0)
+    this = seq_get(args, 1)
+
+    # Every date - time type in tsql resoves to DATETIME2
+    if isinstance(this, exp.Expression) and this.is_string:
```

Comment:
If you switch to `this = args[1]` we can simplify this branch to just:

 ```Python3
if this.is_string:
   ...
```

If we leave `seq_get` in then I'd change this to:

```Python3
if this and this.is_string:
   ...
```

Reply:
Let's not asume `args[1]` will work, e.g. if you set the error level to ignore that'd fail with an index error. I'd go with the latter.

Reply:
Ah yeah that's right, even if an expression requires some args (in the case of `exp.TimestampTrunc`, both `this` and `unit` are required anyways), setting `error_level=sqlglot.ErrorLevel.IGNORE` trumps that.